vines/init.lua

```diff
@@ -314,100 +369,117 @@ if enable_roots ~= false then
-	choose_random_wall = true,
-	neighbors = {
-		"default:jungleleaves",
-		"moretrees:jungletree_leaves_red",
-		"moretrees:jungletree_leaves_yellow",
-		"moretrees:jungletree_leaves_green"
-	},
-	avoid_nodes = {
-		"vines:jungle_middle",
-		"vines:jungle_end",
-	},
-	avoid_radius = 5,
-	spawn_delay = 500,
-	spawn_chance = 100,
-	spawn_surfaces = {
-		"default:jungletree",
-		"moretrees:jungletree_trunk"
-	},
-	spawn_on_side = true,
-	plantlife_limit = -0.9,
-	humidity_min = 0.2,
-})
+if enable_side ~= false then
+	vines.register_vine('side',
+		{description = S("Vines"), average_length = 6}, {
+		choose_random_wall = true,
+		avoid_nodes = {"group:vines", "default:apple"},
+		avoid_radius = 3,
+		surface = {
+	--		"default:leaves",
+			"default:jungleleaves",
+			"moretrees:jungletree_leaves_red",
+			"moretrees:jungletree_leaves_yellow",
+			"moretrees:jungletree_leaves_green"
+		},
+		spawn_on_side = true,
+		plantlife_limit = -0.9,
+		rarity = rarity_side,
+	--	humidity_min = 0.4,
+	})
+else
+	minetest.register_alias('vines:side_middle', 'air')
+	minetest.register_alias('vines:side_end', 'air')
+end
 
-vines.register_vine( 'willow',
-	{description = S("Willow Vines"), average_length = 9}, {
-	choose_random_wall = true,
-	avoid_nodes = {"vines:willow_middle"},
-	avoid_radius = 5,
-	near_nodes = {'default:water_source'},
-	near_nodes_size = 1,
-	near_nodes_count = 1,
-	near_nodes_vertical = 7,
-	plantlife_limit = -0.8,
-	spawn_chance = 10,
-	spawn_delay = 500,
-	spawn_on_side = true,
-	spawn_surfaces = {"moretrees:willow_leaves"},
-	humidity_min = 0.5
-})
+if enable_jungle ~= false then
+	vines.register_vine("jungle",
+		{description = S("Jungle Vines"), average_length = 7}, {
+		choose_random_wall = true,
+		neighbors = {
+			"default:jungleleaves",
+			"moretrees:jungletree_leaves_red",
+			"moretrees:jungletree_leaves_yellow",
+			"moretrees:jungletree_leaves_green"
+		},
+		avoid_nodes = {
+			"vines:jungle_middle",
+			"vines:jungle_end",
+		},
+		avoid_radius = 5,
+		surface = {
+			"default:jungletree",
+			"moretrees:jungletree_trunk"
+		},
+		spawn_on_side = true,
+		plantlife_limit = -0.9,
+		rarity = rarity_jungle,
+	--	humidity_min = 0.2,
+	})
+else
+	minetest.register_alias('vines:jungle_middle', 'air')
+	minetest.register_alias('vines:jungle_end', 'air')
+end
 
+if enable_willow ~= false then
+	vines.register_vine( 'willow',
+		{description = S("Willow Vines"), average_length = 9}, {
+		choose_random_wall = true,
+		avoid_nodes = {"vines:willow_middle"},
+		avoid_radius = 5,
+		near_nodes = {'default:water_source'},
+		near_nodes_size = 1,
+		near_nodes_count = 1,
+		near_nodes_vertical = 7,
+		plantlife_limit = -0.8,
+		spawn_on_side = true,
+		surface = {"moretrees:willow_leaves"},
+		rarity = rarity_willow,
+	--	humidity_min = 0.5
+	})
+else
+	minetest.register_alias('vines:willow_middle', 'air')
+	minetest.register_alias('vines:willow_end', 'air')
+end
 
-print("[Vines] Loaded!")
+print("[Vines] Loaded")
```

Comment:
```suggestion
```
While changing it this could be just removed, engine already gives this information with info logging and also includes time wasted while loading mod.